### PR TITLE
Adding new `sns-topic` module

### PR DESCRIPTION
**Pull request by danjbh**
## what
* Adding new `sns-topic` module

## why
* Because we don't have one yet!

### CODE REVIEW

File: modules/sns-topic/default.auto.tfvars [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file is included by default in terraform plans
+enabled = false
+
+name = "sns-topic"
+
```
Comment (nitrocode) [inline modules/sns-topic/default.auto.tfvars:5]: all of this we can keep in the catalog except for enabled=false

```suggestion
```

File: modules/sns-topic/README.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+**Stack Level**: Regional
+
+Here are some example snippets for how to use this component:
+
+`stacks/catalog/sns-topic/defaults.yaml` file (base component for all SNS topics with default settings):
+
+```yaml
+components:
+  terraform:
+    sns-topic-defaults:
```
Comment (nitrocode) [inline modules/sns-topic/README.md:16]:
```suggestion
    sns-topic/defaults:
```

File: modules/sns-topic/README.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import:
+- catalog/sns-topic/defaults
+
+components:
+  terraform:
+    template-bucket:
+      metadata:
+        component: sns-topic
+        inherits:
+          - sns-topic-defaults
```
Comment (nitrocode) [inline modules/sns-topic/README.md:56]:
```suggestion
          - sns-topic/defaults
```

**Review by nitrocode — COMMENTED** [review]
Nice addition. Please see the comments and then LGTM

File: modules/sns-topic/default.auto.tfvars [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# This file is included by default in terraform plans
+enabled = false
+
+name = "sns-topic"
```
Comment (milldr) [inline modules/sns-topic/default.auto.tfvars:4]:
```suggestion
```

File: modules/sns-topic/README.md (hunk 2 above)
Comment (danjbh) [reply modules/sns-topic/README.md:16]: hey @nitrocode, I was following the latest pattern and using `s3-bucket` as an example, which uses something similar to what I have -- https://github.com/cloudposse/terraform-aws-components/tree/master/modules/s3-bucket

Have a look and let me know what you think.

**Review by nitrocode — DISMISSED** [review]

File: modules/sns-topic/README.md (hunk 2 above)
Comment (nitrocode) [reply modules/sns-topic/README.md:16]: I prefer the forward slash since this creates the nice folder structure in spacelift :)

Comment (nitrocode) [reply modules/sns-topic/README.md:16]: It's also more intuitive IMHO because `sns-topic/defaults` matches the catalog file path

Comment (danjbh) [reply modules/sns-topic/README.md:16]: Fair enough.. I'll make the change.

**Review by nitrocode — APPROVED** [review]

**Pull request merged**